xbmc/windowing/osx/WinEventsOSXImpl.mm

```diff
@@ -195,103 +195,147 @@ - (void)ProcessInputEvent:(NSEvent*)nsEvent
 - (NSEvent*)InputEventHandler:(NSEvent*)nsEvent
 {
   bool passEvent = true;
-  // The incoming mouse position.
-  NSPoint location = nsEvent.locationInWindow;
-  if (!nsEvent.window || location.x < 0 || location.y < 0)
-    return nsEvent;
-
-  location = [nsEvent.window convertPointToBacking:location];
-
-  // cocoa world is upside down ...
-  auto winSystem = dynamic_cast<CWinSystemOSX*>(CServiceBroker::GetWinSystem());
-  if (!winSystem)
-    return nsEvent;
-
-  NSRect frame = [nsEvent.window convertRectToBacking:winSystem->GetWindowDimensions()];
-  location.y = frame.size.height - location.y;
-
   XBMC_Event newEvent = {};
 
   switch (nsEvent.type)
   {
     // handle mouse events and transform them into the xbmc event world
     case NSEventTypeLeftMouseUp:
-      newEvent.type = XBMC_MOUSEBUTTONUP;
-      newEvent.button.button = XBMC_BUTTON_LEFT;
-      newEvent.button.x = location.x;
-      newEvent.button.y = location.y;
-      [self MessagePush:&newEvent];
+    {
+      auto location = [self TranslateMouseLocation:nsEvent];
+      if (location.has_value())
+      {
+        NSPoint locationCoordinates = location.value();
+        newEvent.type = XBMC_MOUSEBUTTONUP;
+        newEvent.button.button = XBMC_BUTTON_LEFT;
+        newEvent.button.x = locationCoordinates.x;
+        newEvent.button.y = locationCoordinates.y;
+        [self MessagePush:&newEvent];
+      }
```

Comment:
you could probably extract this whole logic to a separate function to avoid copy-paste, only take NSEventTypeScrollWheel into account (either return flag or check if `newEvent.type` became valid or provide lambda parameter for further modifications)